scripts/get-workspace-map.sh

```diff
@@ -0,0 +1,92 @@
+#!/bin/bash
+# get-workspace-map.sh [--stage] [--prod] [context-1 context-2... context-n]
+#   Read Namespace resources and generate a JSON object mapping from namespaces
+#   to workspace based on the value of label toolchain.dev.openshift.com/space.
+#
+#   This script assumes `oc` is preconfigured with all the required clusters
+#   and could be found in $PATH.
+#
+#   if --stage or --prod provided, the script will look for contexts matching
+#   staging and production clusters.
+#   Any other argument will be considered as an explicit context name.
+#
+set -o pipefail -o errexit -o nounset
+
+# Parse command line arguments
+EXPLICIT=""
+for i in "$@"
+do
+case $i in
+    --stage)
+    STAGE=true
+    ;;
+    --prod)
+    PROD=true
+    ;;
+    *)
+    EXPLICIT="${i} ${EXPLICIT}"
+    ;;
+esac
+done
+
+STAGE=${STAGE:-""}
+PROD=${PROD:-""}
+
+if [[ -n $STAGE ]]; then
+# Take all non-host staging clusters and keep only their names
+  STAGE=$(
+    oc config get-contexts \
+      | { grep -E 'api-stone-stg\S+' || test $? = 1; } \
+      | { grep -v api-stone-stg-host || test $? = 1; } \
+      | sed 's/^[ \*]*//g' \
+      | sed 's/ .*$//g' \
+      | tr '\n' ' '
+  )
+  if [[ -z $STAGE ]]; then
+    echo "No staging clusters found."
+  fi
+fi
+
+if [[ -n $PROD ]]; then
+# Take all non-host production clusters and keep only their names
+  PROD=$(
+    oc config get-contexts \
+      | { grep -E 'api-stone-prd\S+' || test $? = 1; } \
+      | { grep -v api-stone-prd-host || test $? = 1; } \
+      | sed 's/^[ \*]*//g' \
+      | sed 's/ .*$//g' \
+      | tr '\n' ' '
+  )
+  if [[ -z $PROD ]]; then
+    echo "No production clusters found."
+  fi
+fi
+
+if [[ -z $STAGE && -z $PROD && -z $EXPLICIT ]]; then
+  echo "No clusters found."
+  exit 1
+fi
+
+contexts="${STAGE} ${PROD} ${EXPLICIT}"
+
+out={}
+for context in $contexts
```

Comment:
are you planning to get rid of this `for` in favour of `xargs` and a single pipeline?

Reply:
I'm trying. I wasn't able to make it work yet.

Reply:
What I'm unable to do, at least not in a pretty way, and for something which isn't much simpler than what we're trying to do here, is to combine at output of all the `oc` executions to one `jq` invocation.